jsonrpc/eth_endpoint.go

```diff
@@ -393,23 +393,53 @@ func (e *Eth) GetStorageAt(
 	return argBytesPtr(result), nil
 }
 
-// GasPrice returns the average gas price based on the last x blocks
-// taking into consideration operator defined price limit
+// GasPrice exposes "getGasPrice"'s function logic to public RPC interface
 func (e *Eth) GasPrice() (interface{}, error) {
+	gasPrice, err := e.getGasPrice()
+	if err != nil {
+		return nil, err
+	}
+
+	return argUint64(gasPrice), nil
+}
+
+// getGasPrice returns the average gas price based on the last x blocks
+// taking into consideration operator defined price limit
+func (e *Eth) getGasPrice() (uint64, error) {
 	// Return --price-limit flag defined value if it is greater than avgGasPrice/baseFee+priorityFee
 	if e.store.GetForksInTime(e.store.Header().Number).London {
 		priorityFee, err := e.store.MaxPriorityFeePerGas()
 		if err != nil {
-			return nil, err
+			return 0, err
 		}
 
-		return argUint64(common.Max(e.priceLimit, priorityFee.Uint64()+e.store.GetBaseFee())), nil
+		return common.Max(e.priceLimit, priorityFee.Uint64()+e.store.GetBaseFee()), nil
 	}
 
 	// Fetch average gas price in uint64
 	avgGasPrice := e.store.GetAvgGasPrice().Uint64()
 
-	return argUint64(common.Max(e.priceLimit, avgGasPrice)), nil
+	return common.Max(e.priceLimit, avgGasPrice), nil
+}
+
+// fillTransactionGasPrice fills transaction gas price if no provided
+func (e *Eth) fillTransactionGasPrice(tx *types.Transaction) error {
+	if tx.GetGasPrice(e.store.GetBaseFee()).BitLen() > 0 {
+		return nil
+	}
+
+	estimatedGasPrice, err := e.getGasPrice()
+	if err != nil {
+		return err
+	}
+
+	if tx.Type == types.DynamicFeeTx {
+		tx.GasFeeCap = new(big.Int).SetUint64(estimatedGasPrice)
```

Comment:
Do we need to assign any value to GasTipCap as well for the checks ?

Reply:
Nope, no needed